### PR TITLE
feat: switch publish dryrun required status based on publish settings

### PR DESCRIPTION
**Pull request by loispostula**
resolves https://github.com/ForesightMiningSoftwareCorporation/infrastructure/issues/195

Change logic for dry-run test to be required in the publish context when we want to publsih to docker or private reg

### CODE REVIEW

**Comment by loispostula** [comment]
@greaka I've also taken the opportunity to switch reference to `v1.2.0` to `v1`

**Comment by loispostula** [comment]
And I've tested the behaviour here https://github.com/ForesightMiningSoftwareCorporation/ci_tests/pull/16

File: .github/workflows/rust-build.yml [hunk 1]
```diff
@@ -120,12 +116,12 @@ jobs:
       additional_args: ${{ inputs.additional_args }}
       custom_cargo_commands: ${{ inputs.custom_cargo_commands }}
       login_private_registry: ${{ inputs.publish_public_registry != 'true' }}
-      skip-test-publish: ${{ inputs.skip-test-publish }}
+      test-publish-required: ${{ inputs.publish_docker == 'true' && inputs.publish_binary == 'true' && inputs.publish_binary == 'true' }}
```
Comment (greaka) [inline .github/workflows/rust-build.yml:119]: When are these flags set in the real world? Shouldn't that only be true for runs on the main branch and not already as part of a PR? Also am I missing something or is the same field checked twice here?

**Review by greaka — APPROVED** [review]
Looks good! We should maybe think about removing the summary tag. We collapse old results anyway and it sounded like non-required failures are easily overlooked.

**Comment by loispostula** [comment]
I agree that we can drop the sumary tag. I also think we should convert the reporting to a js action, as it would give us more control. In the `infrastructure` repo, I use this action: https://github.com/ahmadnassri/action-terraform-report which delete its previous comment, it's even clearer than the hiding mechanism we currently uses. I've created https://github.com/ForesightMiningSoftwareCorporation/infrastructure/issues/202 to track

File: .github/workflows/rust-build.yml (hunk 1 above)
Comment (loispostula) [reply .github/workflows/rust-build.yml:119]: Currently we used them here https://github.com/search?q=org%3AForesightMiningSoftwareCorporation+skip-test-publish++NOT+repo%3AForesightMiningSoftwareCorporation%2Fgithub&type=code but I agree that we should maybe not make them available to enforce at org level

Comment (loispostula) [reply .github/workflows/rust-build.yml]: Thanks for the typo find, I've fixed it

File: .github/workflows/rust-build.yml [hunk 2]
```diff
@@ -120,12 +116,12 @@ jobs:
       additional_args: ${{ inputs.additional_args }}
       custom_cargo_commands: ${{ inputs.custom_cargo_commands }}
       login_private_registry: ${{ inputs.publish_public_registry != 'true' }}
-      skip-test-publish: ${{ inputs.skip-test-publish }}
+      test-publish-required: ${{ inputs.publish_docker == 'true' && inputs.publish_private_registry == 'true' && inputs.publish_public_registry == 'true' }}
```
Comment (greaka) [inline .github/workflows/rust-build.yml:119]: Is `&&` the correct operation here?

Comment (loispostula) [reply .github/workflows/rust-build.yml]: :o no, thanks for the catch, again

**Review by greaka — APPROVED** [review]

**Pull request merged**